homeassistant/components/rainmachine/util.py

```diff
@@ -35,6 +37,39 @@ class RunStates(StrEnum):
 }
 
 
+@callback
+def async_clean_up_old_entities(
+    hass: HomeAssistant, entry: ConfigEntry, unique_id_suffixes_to_remove: Iterable[str]
+) -> None:
+    """Clean up old, no-longer-used entities."""
+    ent_reg = entity_registry.async_get(hass)
+    for entity_registry_item in [
+        e
+        for e in ent_reg.entities.values()
+        if e.config_entry_id == entry.entry_id
+        and any(
+            suffix
+            for suffix in unique_id_suffixes_to_remove
+            if e.unique_id.endswith(suffix)
+        )
+    ]:
+        removed_entity_id = entity_registry_item.entity_id
+        async_create_issue(
+            hass,
+            DOMAIN,
+            f"removed_old_entity_{removed_entity_id}",
+            is_fixable=True,
```

Comment:
We should only set `is_fixable` if we provide a config flow. What would this config flow do?

Reply:
This came about from a conversation with @frenck. This triggers the [default implementation of `ConfirmRepairFlow`](https://github.com/home-assistant/core/blob/7d0a4ee00a5af7bbb1ecc22a031285557e9be76b/homeassistant/components/repairs/issue_handler.py#L74-L75). As I understand it, using the default `ConfirmRepairFlow` in this manner is a way to tell the user, "Hey, don't ignore this – you need to explicitly confirm that you understand what this issue indicates."

@frenck, do I have that right?

Reply:
> We should only set is_fixable if we provide a config flow. What would this config flow do?

Setting it without implementing the flow in a platform triggers the build-in confirm-only flow.

But honestly, I'm not sure if raising a repairs issue, in this case, is granted. For example, do we not have to remove the sensor? You could disable it by default?

Reply:
That doesn't feel right. It's going to be disabled forever, which effectively orphans it.

I guess I'm still confused on _when_ to create a repairs issue. I chose to here because I could explicitly tell the user the entity ID of the entity being replaced (and of its replacement).